scripts/Jenkinsfile

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env groovy
+
+import java.text.SimpleDateFormat
+
+//===----------------------------------------------------------------------===//
+// Helpers.                                                                   //
+//===----------------------------------------------------------------------===//
+
+def git_config() {
+  def git_email = "rocmhcc@amd.com" // "jenkins-compute@amd.com"
+  def git_name = "ROCm HCC" // Jenkins Compute"
+
+  sh (script:"git config user.email ${git_email}")
+  sh (script:"git config user.name ${git_name}")
+}	
+
+def git_remote_add(workspace, remote_name, remote_url) {
+   try {
+	    dir (workspace) {
+		  sh (script:"git remote add ${remote_name} ${remote_url}")
+		}
+	} catch (e) { 
+		sh (script:"echo \"fork already exists\"")
+	}
+}
+ 
+ def git_merge(workspace, merge_name, checkout_branch, merge_branch) {
+ 	try {
+ 		dir(workspace) {
+ 			git_config()
+	    	sh "git fetch --all"
+	    	sh "git checkout -b ${merge_name} ${checkout_branch}"
+	    	sh "git merge ${ merge_branch} --no-edit"
+    	}   	
+		
+    } catch (e) {
+    	sh (script: "echo \"GIT MERGE failed\"")
+    	throw e
+    }
+ }
+ 
+def send_email(status) {
+  emailext (
+    to: 'rocm.hcc@amd.com',
+    subject: "${status}: Job '${env.JOB_NAME} [${env.BUILD_NUMBER}]'",
+    body: "${status}: Job '${env.JOB_NAME} [${env.BUILD_NUMBER}]'; see console output at ${env.BUILD_URL}"
+  )
+}
+
+node ('upstream-merger')   
+{
+  try { 
+	     def dateFormat = new SimpleDateFormat("yyyyMMddHH")
+		 def date = new Date()
+		 def nproc = 8
```

Comment:
@david-salinas for `nproc` , what I meant was to run `nproc` program on the system to determine available cores, instead of fixing a value. I'm not well-versed in jenkins file, is it possible to it within jenkine?